daemon/cmd/deletion_queue.go

```diff
@@ -26,26 +27,39 @@ var deletionQueueCell = cell.Group(
 type deletionQueue struct {
 	lf            *lockfile.Lockfile
 	daemonPromise promise.Promise[*Daemon]
+	wg            sync.WaitGroup
 }
 
-func (dq *deletionQueue) Start(ctx cell.HookContext) error {
-	d, err := dq.daemonPromise.Await(ctx)
-	if err != nil {
-		return err
-	}
+func (dq *deletionQueue) Start(cell.HookContext) error {
+	dq.wg.Add(1)
+	go func() {
+		defer dq.wg.Done()
 
-	if err := dq.lock(ctx); err != nil {
-		return err
-	}
+		// hook context cancels when the start hooks have run, use context.Background()
+		// as we may be running after that.
+		d, err := dq.daemonPromise.Await(context.Background())
```

Comment:
Please use the context provided by `Start()` here

Reply:
Ah, I missed the comment sorry. We have lots of other code still using the start `ctx` for  `dq.daemonPromise.Await`. You need to fix those up as well then I guess?

Reply:
This change here also seems to change the behavior. We're no longer handling the error from `processQueuedDeletes` and continuing with starting when earlier we stopped if it failed. This also runs now asynchronously with everything else rather than sequentially during start-up. Does this break something?

And yeah, as @gandro said, all `Await`s for daemon promise will now need to use a context that lives beyond `Start`.

Reply:
Checked again, no start context use in daemon promise `Await()` any more.